src/test/java/com/ericsson/ei/notifications/EmailSenderTest.java

```diff
@@ -43,14 +44,33 @@
     @InjectMocks
     private EmailSender emailSender;
 
+    @Value("${email.sender:noreply@ericsson.com}")
+    private String sender;
+
+    @Value("${email.subject:Email notification}")
+    private String subject;
+
     @Test(expected = NotificationFailureException.class)
     public void sendEmailThrowsException() throws Exception {
         doThrow(new MailSendException("")).when(javaMailSender).send(message);
         emailSender.sendEmail(message);
     }
 
+    @Test
+    public void testDefaultValues() {
```

Comment:
I think this test can be removed since it currently does not seem to be testing anything.

Reply:
Emelie has asked for the test case, so I have added the change.

Reply:
You could make an instance of EmailSender and then get SENDER and SUBJECT and make sure they are set.
Then you are testing the logic through the EmailSender class.
There is also no assertion in the test so it would accept any outcome. A test should include an assert that checks some expected outcome of the test, unless this for reason is not possible. In your case you instantiate the class, get the values and assert that they were set to the default values.
You can also check that you can also override the defaults as expected by setting the environment variables first and then make an EmailSender instance and check that SENDER and SUBJECT were indeed set to something other than default.

Reply:
After seeing the funcitonal test scenario added above, I agree that this test is not really needed. This is basically testing that Spring has assigned the correct values for the properties, and we want to test our logic - not Spring's. If the annotations are used correctly (see above comments by @a-breid-ericsson and @Christoffer-Cortes ) only Spring assigns the value to the variables and we don't have to do this in the EmailSender class. But we do want to check that if no configuration was given we use the default values - and this is tested in the functional tests.